src/shogun/util/factory.h

```diff
@@ -267,5 +272,28 @@ namespace shogun
 		SG_REF(result);
 		return result;
 	}
+
+	/** Create a pipeline builder.
+	 * See also CPipelineBuilder and CPipeline.
+	 * @return new instance of CPipelineBuilder
+	 */
+	CPipelineBuilder* pipeline()
+	{
+		auto result = new CPipelineBuilder();
+		SG_REF(result);
+		return result;
+	}
+
+	CMachineEvaluation* machine_evaluation(const std::string& name, CMachine* machine)
+	{
+		auto obj = create_object<CMachineEvaluation>(name.c_str());
+		obj->put("machine", machine);
+		return obj;
+	}
+
+	CMachineEvaluation* machine_evaluation(CSGObject* obj)
```

Comment:
I am not so keen on this one....but did you actually use it?

Reply:
no...but every other factory has this, do we need it?

Reply:
Ah! Of course, you can actually even use this as the "as_machine"